### PR TITLE
15 use search opts in android

### PR DESCRIPTION
**Pull request by MauByr**

### CODE REVIEW

**Comment by MauByr** [comment]
depends on #5

**Comment by JasBuc** [comment]
> depends on #5

Do you mean it is branched off of #5 ? If so can we point the review at #5's branch rather than master please?

File: SharedCode/src/commonMain/kotlin/dataObjects/frontendDataObjects/JourneyTableDataElem.kt [hunk 1]
```diff
@@ -2,45 +2,75 @@ package com.jetbrains.handson.mpp.mobile.dataObjects.frontendDataObjects
...
 
 }
 
 class JourneyStation(station: Station) {
+    constructor(stationInfo: StationInfo) : this(
+        Station(
+            stationInfo.name ?: "Unknown",
+            stationInfo.crs ?: "",
+            stationInfo.nlc ?: "000"
+        )
```
Comment (JasBuc) [inline SharedCode/src/commonMain/kotlin/dataObjects/frontendDataObjects/JourneyTableDataElem.kt:67]: Like I commented elsewhere, we might just want to make them null in this case?

File: SharedCode/src/commonMain/kotlin/dataObjects/frontendDataObjects/JourneyTableDataElem.kt [hunk 2]
```diff
@@ -2,45 +2,75 @@ package com.jetbrains.handson.mpp.mobile.dataObjects.frontendDataObjects
...
+    )
+
     val shortName: String = station.displayName.replace("[aeiou]".toRegex(), "")
     val fullName: String = station.displayName
     val crsCode: String = station.crs
+    override fun toString(): String {
+        return fullName
+    }
```
Comment (JasBuc) [inline SharedCode/src/commonMain/kotlin/dataObjects/frontendDataObjects/JourneyTableDataElem.kt:75]: Why do we need this, can't we just reference fullName where this is used?

File: SharedCode/src/commonMain/kotlin/dataObjects/frontendDataObjects/JourneyTableDataElem.kt [hunk 3]
```diff
@@ -2,45 +2,75 @@ package com.jetbrains.handson.mpp.mobile.dataObjects.frontendDataObjects
...
 
+    @Deprecated(level = DeprecationLevel.WARNING, message = "Use endTime Property instead")
     fun getArrivalTime(): String {
         return DateFormat("HH:mm").format(endTimeRaw)
     }
 
+    @Deprecated(level = DeprecationLevel.WARNING, message = "Use journeyTime Property instead")
     fun getJourneyTime(): String {
-        return "${journeyTime / 60}:${journeyTime % 60}"
+        return "${journeyTimeMins / 60}:${journeyTimeMins % 60}"
```
Comment (JasBuc) [inline SharedCode/src/commonMain/kotlin/dataObjects/frontendDataObjects/JourneyTableDataElem.kt:56]: Can we delete these getJourneyTime() getArrivalTime() getStartTime etc. functions and just reference e.g. it.journeyTime?

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt [hunk 4]
```diff
@@ -28,18 +27,28 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
...
     }
 
     private fun setupSearchListener(presenter: ApplicationPresenter) {
         val button: Button = findViewById(R.id.search_button)
-        button.setOnClickListener { presenter.onSearchClicked(timeUTCString = null) }
+        button.setOnClickListener {
+            val fromStation = departureSpinner?.selectedItem as JourneyStation?
+            val toStation = arrivalSpinner?.selectedItem as JourneyStation?
+            if(fromStation != null && toStation!=null){
+                presenter.onSearchClicked(fromStation.crsCode,toStation.crsCode,timeUTCString = null)
```
Comment (JasBuc) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt:48]: Nit: spacing between item here and in line above. Also excess new line spacing below

File: app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt [hunk 5]
```diff
@@ -28,18 +27,28 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
     }
 
     override fun showResults(result: List<JourneyTableDataElem>) {
-        findViewById<TextView>(R.id.search_results).text = result.toString()
+        findViewById<RecyclerView>(R.id.search_results).adapter = SearchResultAdapter(result)//.text = result.toString()
```
Comment (JasBuc) [inline app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt:30]: I think I commented on this elsewhere too, perhaps some rebasing needs to be done to stop these changes showing up twice

**Review by JasBuc — COMMENTED** [review]
Looks like this contains some duplicate changes to another PR, perhaps needs a rebase to avoid this? Looking good though!

**Pull request merged**